File: app/schemas/submission.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.evaluation import EvaluationResponse
from app.utils.enums import SubmissionStatus


class SubmissionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    student_id: int
    test_id: int
    answer_text: str | None
    extracted_text: str | None
    image_url: str | None
    status: SubmissionStatus
    created_at: datetime
    evaluation: EvaluationResponse | None = None

File: eldas-frontend/src/types/submission.ts
import type { Evaluation } from "@/types/evaluation";

export type SubmissionStatus = "pending" | "processing" | "completed" | "failed";

export interface Submission {
  id: number;
  student_id: number;
  test_id: number;
  answer_text: string | null;
  extracted_text: string | null;
  image_url: string | null;
  status: SubmissionStatus;
  created_at: string;
  evaluation?: Evaluation | null;
}
